cmd/worker_pool.go

```diff
@@ -0,0 +1,74 @@
+package cmd
+
+import bosherr "github.com/cloudfoundry/bosh-utils/errors"
+
+type WorkerPool struct {
+	WorkerCount int
+}
+
+// Runs the given set of tasks in parallel using the configured number of worker go routines
+// Will stop adding new tasks if a task throws an error, but will wait for in-flight tasks to finish
+func (w WorkerPool) ParallelDo(tasks ...func() (interface{}, error)) ([]interface{}, error) {
```

Comment:
@ljfranklin found this: https://github.com/cloudfoundry/workpool referenced from dns-release code. usable here? cc @luan